aws/utils/time_sensitive.h

```diff
@@ -63,6 +63,14 @@ class TimeSensitive : private boost::noncopyable {
     set_deadline(Clock::now() + ms);
   }
 
+  void extend_deadline_from_now(std::chrono::milliseconds ms) {
+    if (deadline().time_since_epoch().count() == 0) {
+      set_deadline_from_now(ms);
```

Comment:
Why is this necessary?

Reply:
This is the test used through out the file to see if a particular timepoint is set. I will refactor this out to a method that says so and can be reused.